internal/registry/registry_cache.go

```diff
@@ -63,54 +62,73 @@ func NewRegistryCache(logger logging.Logger, home, registryURL string) (Cache, e
 	}, nil
 }
 
-func NewDefaultRegistryCache(logger logging.Logger, home string) (Cache, error) {
-	return NewRegistryCache(logger, home, defaultRegistryURL)
-}
-
-func (r *Cache) createCache() error {
-	r.logger.Debugf("Creating registry cache for %s/%s", r.url.Host, r.url.Path)
-
-	root, err := ioutil.TempDir("", "registry")
+// LocateBuildpack stored in registry
+func (r *Cache) LocateBuildpack(bp string) (Buildpack, error) {
+	err := r.Refresh()
 	if err != nil {
-		return err
+		return Buildpack{}, errors.Wrap(err, "refreshing cache")
 	}
 
-	repository, err := git.PlainClone(root, false, &git.CloneOptions{
-		URL: r.url.String(),
-	})
+	ns, name, version, err := buildpack.ParseRegistryID(bp)
 	if err != nil {
-		return errors.Wrap(err, "could not clone remote registry")
+		return Buildpack{}, errors.Wrap(err, "parsing buildpacks registry id")
 	}
 
-	w, err := repository.Worktree()
+	entry, err := r.readEntry(ns, name)
 	if err != nil {
-		return err
+		return Buildpack{}, errors.Wrap(err, "reading entry")
 	}
 
-	return os.Rename(w.Filesystem.Root(), r.Root)
+	if len(entry.Buildpacks) > 0 {
+		if version == "" {
+			highestVersion := entry.Buildpacks[0]
+			if len(entry.Buildpacks) > 1 {
+				for _, bp := range entry.Buildpacks[1:] {
+					if semver.Compare(fmt.Sprintf("v%s", bp.Version), fmt.Sprintf("v%s", highestVersion.Version)) > 0 {
+						highestVersion = bp
+					}
+				}
+			}
+			return highestVersion, highestVersion.Validate()
+		}
+
+		for _, bpIndex := range entry.Buildpacks {
+			if bpIndex.Version == version {
+				return bpIndex, bpIndex.Validate()
+			}
+		}
+		return Buildpack{}, fmt.Errorf("could not find version for buildpack: %s", bp)
```

Comment:
```suggestion
		return Buildpack{}, fmt.Errorf("finding version for buildpack: %s", bp)
```

Reply:
This I didn't change; this was a generated error, so I figured that using the `could not find` syntax would be a bit clearer.